llama_hub/confluence/base.py

```diff
@@ -289,7 +289,9 @@ def process_page(self, page, include_attachments, text_maker):
             attachment_texts
         )
         return Document(
-            text=text, doc_id=page["id"], extra_info={"title": page["title"]}
+            text=text,
+            doc_id=page["id"],
+            extra_info={"title": page["title"], "page_id": page["id"]},
```

Comment:
We already have the page_id on the doc_id, can't you use this one instead of add a new field on extra_info?

```python
# Specify the document IDs you want to filter
doc_ids = ["doc_id_1", "doc_id_2"]

# Build the retriever with the doc_ids parameter
retriever = VectorIndexRetriever(
    index=index,
    similarity_top_k=3,
    vector_store_query_mode="default",
    doc_ids=doc_ids,
)
```

Reply:
@EmanuelCampos  Right now response I get back includes page id inside but logic is complicated. I'm trying to ensure title and page id are available together so constructing the reference hyperlink is easier to cite properly using metadata. I have workaround but its not as clean as it should with out extra_info

```
                for s in response.source_nodes:
                    tempdict = s.node.relationships
                    pageid = tempdict[
                        NodeRelationship.SOURCE.value].node_id
                    pagelinks.add(f"https://wiki.kbohra.com/pages/viewpage.action?pageId={pageid}")
```

response.source_nodes
```
response source nodes: [NodeWithScore(node=TextNode(id_='01e24260-c604-4d17-9db2-7f9afee74f82', embedding=None, metadata={'title': 'Incident Diagonistics?'}, excluded_embed_metadata_keys=[], excluded_llm_metadata_keys=[], relationships={<NodeRelationship.SOURCE: '1'>: RelatedNodeInfo(node_id='2692275965', node_type=None, metadata={'title': 'Incident Diagonistics?'}, hash='da520f8b2c4b43b81fc76034acad3c88404e1ad27c82ddf3ca552c7db56f2baa')}
```